### PR TITLE
Updated tutorial for transient PDEs

### PR DESCRIPTION
**Pull request by AlexandreMagueresse**
The tutorials needed to be updated following the refactoring of the ODE module of Gridap. I have edited the previous tutorial on the heat equation and added a new tutorial on a nonlinear variant of the heat equation (where the diffusivity coefficient is polynomial in `u`) to illustrate the difference in the description of the ODE operator.

### CODE REVIEW

**Comment by JordiManyer** [comment]
@amartinhuertas Could we approve this?

**Comment by amartinhuertas** [comment]
> @amartinhuertas Could we approve this?

Is the PR ready to be merged?

**Comment by AlexandreMagueresse** [comment]
The versions of `Gridap` and `GridapMakie` may need to be checked or updated in the project file. (Last time I checked, all the tutorials were compatible with the last version of `GridapMakie`). The rest is ready to be merged.

File: Project.toml
Comment (alex-fu27) [inline Project.toml:1]: This wont produce a satisfiable dependency graph, to get `develop .` to accept it I have to weaken the versions of `GridapGmsh`, `GridapP4est` and remove the `GridapMakie` dependency.

Comment (alex-fu27) [reply Project.toml:1]: And also `GridapEmbedded` to get `instantiate` to work.

**Comment by JordiManyer** [comment]
Can we merge this now that GridapMakie has been updated?

**Comment by amartinhuertas** [comment]
Yes. Accepting ...

**Pull request merged**